test/Numerics/DGmethods/Euler/acousticwave_mrigark.jl

```diff
@@ -0,0 +1,406 @@
+struct EarthParameterSet <: AbstractEarthParameterSet end
+const param_set = EarthParameterSet()
+
+using MPI, Logging, StaticArrays, LinearAlgebra, Printf, Dates, Test
+
+const output_vtk = false
+
+"""
+    main()
+
+Run this test problem
+"""
+function main()
+    ClimateMachine.init()
+    ArrayType = ClimateMachine.array_type()
+
+    mpicomm = MPI.COMM_WORLD
+
+    polynomialorder = 5
+    numelem_horz = 10
+    numelem_vert = 5
+
+    # Real test should be run for 33 hour, which is approximate time for the
+    # pulse to go around the whole sphere
+    # but for CI we only run 1 hour
+    # timeend = 60 * 60 * 33
+    timeend = 60 * 60
+
+    # Do the output every hour
+    outputtime = 60 * 60
+
+    # Expected result is L2-norm of the final solution
+    expected_result = Dict()
+    expected_result[Float64, true] = 9.5073337869322578e+13
+    expected_result[Float64, false] = 9.5073455070673703e+13
+
+    @testset "acoustic wave" begin
+        for FT in (Float64,)# Float32)
+            for explicit in (true, false)
+                result = run(
+                    mpicomm,
+                    polynomialorder,
+                    numelem_horz,
+                    numelem_vert,
+                    timeend,
+                    outputtime,
+                    ArrayType,
+                    FT,
+                    explicit,
+                )
+                @test result ≈ expected_result[FT, explicit]
+            end
+        end
+    end
+end
+
+"""
+    run(
+        mpicomm,
+        polynomialorder,
+        numelem_horz,
+        numelem_vert,
+        timeend,
+        outputtime,
+        ArrayType,
+        FT,
+        explicit_solve,
+    )
+
+Run the actual simulation.
+"""
+function run(
+    mpicomm,
+    polynomialorder,
+    numelem_horz,
+    numelem_vert,
+    timeend,
+    outputtime,
+    ArrayType,
+    FT,
+    explicit_solve,
+)
+
+    # Structure to pass around to setup the simulation
+    setup = AcousticWaveSetup{FT}()
+
+    # Create the cubed sphere mesh
+    _planet_radius::FT = planet_radius(param_set)
+    vert_range = grid1d(
+        _planet_radius,
+        FT(_planet_radius + setup.domain_height),
+        nelem = numelem_vert,
+    )
+    topology = StackedCubedSphereTopology(mpicomm, numelem_horz, vert_range)
+
+    grid = DiscontinuousSpectralElementGrid(
+        topology,
+        FloatType = FT,
+        DeviceArray = ArrayType,
+        polynomialorder = polynomialorder,
+        meshwarp = cubedshellwarp,
+    )
+    hmnd = min_node_distance(grid, HorizontalDirection())
+    vmnd = min_node_distance(grid, VerticalDirection())
+
+    # This is the base model which defines all the data (all other DGModels
+    # for substepping components will piggy-back off of this models data)
+    fullmodel = AtmosModel{FT}(
+        AtmosLESConfigType,
+        param_set;
+        orientation = SphericalOrientation(),
+        ref_state = HydrostaticState(IsothermalProfile(setup.T_ref), FT(0)),
```

Comment:
Just a heads up, you’ll need to rebase this (and the other driver) if #1057 lands

Reply:
Thanks.

Reply:
Thanks. Fix this now.